src/operator/tensor/dot-inl.h

```diff
@@ -235,13 +235,21 @@ inline bool DotForwardInferStorageType(const nnvm::NodeAttrs& attrs,
                                      DispatchMode::kFComputeEx);
   }
   if (!dispatched && lhs_stype == kDefaultStorage && rhs_stype == kCSRStorage &&
-      !param.transpose_a && !param.transpose_b) {
```

Comment:
I thought we will add a stype hint argument which defaults to None?

Reply:
Working on that now.